### PR TITLE
[MSVCRT][CRT_APITEST] Implement _wsystem

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Make an ReactOS's CRT function working.

JIRA issue: N/A

## Proposed changes

- Implement `msvcrt!_wsystem` function, by referring `msvcrt!system` function.
- Improve `system()` function.
- Use `WaitForSingleObject` in `system()` and `_wsystem()`.
- Check existence of `COMSPEC`.

### CODE REVIEW

**Comment by tkreuzer** [comment]
Maybe we should just sync this with wine?

File: sdk/lib/crt/process/_system.c [hunk 1]
```diff
@@ -118,6 +119,99 @@ int system(const char *command)
...
+            return 1;
+    }
+
+    if (szComSpec == NULL)
+        return -1;
+
+    // should return 127 or 0 ( MS ) if the shell is not found
+    // _set_errno(ENOENT);
+
+    if (szComSpec == NULL)
```
Comment (JoachimHenze) [inline sdk/lib/crt/process/_system.c:150]: This can never be true, as you do check that a few lines above already and return then. (spotted by ChatGPT)

Comment (katahiromz) [reply sdk/lib/crt/process/_system.c:144]: Fixed.

**Comment by katahiromz** [comment]
Win2k3 `system`:
win2k3-system
Successful.

Win2k3 `_wsystem`:
win2k3-_wsystem
Successful.

Win10 `system`:
win10-system
Successful.

Win10 `_wsystem`:
win10-_wsystem
3 failures.

**Comment by katahiromz** [comment]
AFTER:
VirtualBox_ReactOS_05_03_2023_11_30_25
Successful.

**Comment by katahiromz** [comment]
Win2k3:
win2k3
Successful.

Win10:
image
6 + 9 failures.

**Comment by katahiromz** [comment]
AFTER:
after
Successful.

Let's go!

**Pull request merged**